graph/xml_parser.py

```diff
@@ -204,23 +208,108 @@ def _operator_as_child(self, value, scan):
         out = dict(
             operator=value['operator'],
             negate=value['negate'],
-            result=value['negate'],
+            result=value['result'],
+            comment=value['comment'],
             node=[])
         for child in value['node']:
             if 'operator' in child:
                 out['node'].append(self._operator_as_child(child, scan))
             elif 'extend_definition' in child:
                 out['node'].append(
                     self._find_definition_by_id(
-                        scan, child['extend_definition'], child['negate']))
+                        scan,
+                        child['extend_definition'],
+                        child['negate'],
+                        child['comment']))
             elif 'value_id' in child:
                 out['node'].append(child)
             else:
                 raise ValueError('error - unknown child')
         return out
 
-    def _find_definition_by_id(self, scan, id, negate_status):
+    def _find_definition_by_id(self, scan, id, negate_status, comment):
         for definition in scan['definitions']:
             if definition['id'] == id:
                 definition['node'][0]['negate'] = negate_status
+                definition['node'][0]['comment'] = comment
                 return self._operator_as_child(definition['node'][0], scan)
+
+    def deeper_in_criteria_comments(self, criteria):
+        if criteria.get('operator') is None:
```

Comment:
You can construct the comments dict as you do in the `else` branch, and if `comments['operator']` is None, make it `AND` - this saves some space.